### PR TITLE
Add ability to reference to original dependency in an update sample

### PR DESCRIPTION
**Pull request by c2thorn**
This PR addresses 2 issues:

1. references with underscores ignored
2. ~update samples would overwrite the primary dependency in the original non-update sample. Solved by creating a deep copy of the dependency list~
3. DCL update json can reference its original pre-updated json file. See https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/services/google/compute/samples/update_organization.firewall_policy.json
I added the original primary dependency as a "ghost" dependency where it exists in the list but isn't used for HCL. Works, but makes sample generation even more complex. I can see the argument to push back on the DCL for making this type of reference if you feel that is necessary.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
`DCL update json can reference its original pre-updated json file` 😞

**Review by ScottSuarez — APPROVED** [review]
Added a suggested change. A comment for why the skipping is needed might be helpful for future readers :)

File: tpgtools/sample.go [hunk 1]
```diff
@@ -249,6 +256,12 @@ func (s *Sample) EnumerateWithUpdateSamples() []Sample {
 			var newDeps []Dependency
 			newDeps = append(newDeps, newSample.DependencyList...)
 			newDeps[0] = newSample.generateSampleDependencyWithName(*newSample.PrimaryResource, "primary")
+
+			// add the old primary resource to the list
+			oldPrimaryResource := newSample.generateSampleDependency(*s.PrimaryResource)
```
Comment (ScottSuarez) [inline tpgtools/sample.go:261]: won't this resource also need to have the name 'primary` so the reference that replaces it is correct?

```suggestion
			oldPrimaryResource := newSample.generateSampleDependencyWithName(*s.PrimaryResource, "primary")
```

Comment (c2thorn) [reply tpgtools/sample.go:261]: it shouldn't matter either way, `primary` here is just used as the HCL block name. Like `resource "google_compute_firewall_policy" "primary" {`. Maybe there's a better way to indicate what it does.

Reference substitution looks for the filename: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/tpgtools/sample.go#L182

Comment (c2thorn) [reply tpgtools/sample.go:261]: talked offline. @ScottSuarez's right, this won't work as a reference

**Comment by ScottSuarez** [comment]
updates are pretty complicated after talking with Nat 
https://github.com/hashicorp/terraform-provider-google/issues/9811

**Pull request closed without merging**